### PR TITLE
Improved map import for singleplayer missions

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/9535.
Resolves the file not found errors: http://www.ppmforums.com/viewtopic.php?p=545317#545317

### CODE REVIEW

**Comment by pchote** [comment]
Hacking our file search paths is not very nice.  Please make this use `File.Read` instead of `modData.DefaultFileSystem.Open` instead.

Funnily enough I predicted this issue in my review for #10404.  I regret not insisting on this change back then.

File: OpenRA.Mods.Common/UtilityCommands/ImportLegacyMapCommand.cs [hunk 1]
```diff
@@ -271,7 +280,15 @@ void LoadWaypoints(IniSection waypointSection)
 						new OwnerInit("Neutral")
 					};
 
-					Map.ActorDefinitions.Add(new MiniYamlNode("waypoint" + kv.First, ar.Save()));
+					var waypointName = "waypoint" + kv.First;
+					if (kv.First == 25)
```
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/ImportLegacyMapCommand.cs]: What's the source of these numbers?  Are you sure they are the same for both TD and RA?

Comment (Mailaender) [inline OpenRA.Mods.Common/UtilityCommands/ImportLegacyMapCommand.cs]: Source:

* https://github.com/OpenRA/OpenRA/wiki/C&C%20Singleplayer%20Maps
* http://cnc-comm.com/files/RACG110.TXT

No, they aren't. Should be fixed now.

**Comment by Mailaender** [comment]
Removed the virtual file system just like https://github.com/OpenRA/OpenRA/pull/10922 did.

File: OpenRA.Mods.Cnc/ImportTiberianDawnLegacyMapCommand.cs [hunk 2]
```diff
@@ -143,10 +143,22 @@ public override void LoadPlayer(IniFile file, string section)
...
 
 			ReadOverlay(file);
 		}
+
+		public override void SaveWaypoint(int waypointNumber, ActorReference waypointReference)
+		{
+			var waypointName = "waypoint" + waypointNumber;
+			if (waypointNumber == 25)
+				waypointName = "DefaultFlareLocation";
+			if (waypointNumber == 26)
```
Comment (Phrohdoh) [inline OpenRA.Mods.Cnc/ImportTiberianDawnLegacyMapCommand.cs]: `else if`

Comment (Mailaender) [inline OpenRA.Mods.Cnc/ImportTiberianDawnLegacyMapCommand.cs]: True. Added.

**Comment by pchote** [comment]
Ping @Biofreak1987, @abcdefg30.

**Comment by obrakmann** [comment]
Looks good to me, :+1:

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark:

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/c67fbe7ffa1fffe29a9038425762415ab90e3774...97700196e94082f2917ab8d85f1a849f9f0e7024)

**Pull request merged**